src/quo2/components/wallet/progress_bar/view.cljs

```diff
@@ -1,12 +1,49 @@
 (ns quo2.components.wallet.progress-bar.view
-  (:require [quo2.components.wallet.progress-bar.style :as style]
-            [quo2.theme :as quo.theme]
-            [react-native.core :as rn]))
+  (:require [react-native.core :as rn]
+            [quo2.foundations.colors :as colors]
+            [quo2.theme :as theme]
+            [utils.number]))
 
-(defn- view-internal
-  [props]
+(defn get-border-color
+  [network-state]
+  (let [current-theme (theme/get-theme)]
+    (cond
+      (= current-theme :dark)
+      (cond
+        (= network-state "pending") colors/neutral-70
+        :else                       colors/white-opa-5) ; Add an :else to handle other cases
+      :else colors/neutral-80-opa-5))) ; Add an :else to handle the case when current-theme is not :dark
+
+(defn get-bg-color
+  [network-state]
+  (let [current-theme (theme/get-theme)]
+    (cond
+      (= current-theme :light)
+      (cond
+        (= network-state "pending")   colors/neutral-5
+        (= network-state "confirmed") colors/success-50
+        (= network-state "finalised") (colors/custom-color-by-theme :blue 50 60)
+        (= network-state "error")     colors/danger-50)
+      :else
+      (cond
+        (= network-state "pending")   colors/neutral-80
+        (= network-state "confirmed") colors/success-60
+        (= network-state "finalised") (colors/custom-color-by-theme :blue 60 60)
+        (= network-state "error")     colors/danger-60))))
+
+(defn progress-bar
+  [{:keys [network-state width height marginHorizontal]}]
   [rn/view
    {:accessibility-label :progress-bar
-    :style               (style/root-container props)}])
+    :padding             0}
+   [rn/view
+    {:style {:width             (utils.number/parse-int width)
```

Comment:
Please move styles to its own `style.cljs` file, we recently updated our documentation to capture this guideline : https://github.com/status-im/status-mobile/blob/develop/src/quo2/README.md#directory-structure-and-file-names